v2/api/network/customizations/private_endpoints_extensions.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package customizations
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+
+	network "github.com/Azure/azure-service-operator/v2/api/network/v1beta20220701storage"
+	"github.com/Azure/azure-service-operator/v2/internal/genericarmclient"
+	"github.com/Azure/azure-service-operator/v2/internal/util/kubeclient"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/extensions"
+)
+
+var _ extensions.PostReconciliationChecker = &PrivateEndpointExtension{}
+
+func (extension *PrivateEndpointExtension) PostReconcileCheck(
+	_ context.Context,
+	obj genruntime.MetaObject,
+	_ genruntime.MetaObject,
+	_ kubeclient.Client,
+	_ *genericarmclient.GenericClient,
+	_ logr.Logger,
+	_ extensions.PostReconcileCheckFunc) (extensions.PostReconcileCheckResult, error) {
+
+	endpoint, ok := obj.(*network.PrivateEndpoint)
+	if !ok {
+		return extensions.PostReconcileCheckResult{},
+			errors.Errorf("cannot run on unknown resource type %T, expected *network.PrivateEndpoint", obj)
+	}
+
+	// Type assert that we are the hub type. This will fail to compile if
+	// the hub type has been changed but this extension has not
+	var _ conversion.Hub = endpoint
+
+	// We want to check `ManualPrivateLinkServiceConnections` as these are the ones which are not auto-approved.
+	if connections := endpoint.Status.ManualPrivateLinkServiceConnections; connections != nil {
+		for _, connection := range connections {
+			if *connection.PrivateLinkServiceConnectionState.Status != "Approved" {
+				// Returns 'conditions.NewReadyConditionImpactingError' error
+				return extensions.PostReconcileCheckResultFailure(
```

Comment:
Would it make sense to list all the connections that need approval in one go?

Reply:
Updated!